lib/glific/templates/interactive_templates.ex

```diff
@@ -113,17 +113,77 @@ defmodule Glific.Templates.InteractiveTemplates do
   @spec create_interactive_template(map()) ::
           {:ok, InteractiveTemplate.t()} | {:error, Ecto.Changeset.t()}
   def create_interactive_template(attrs) do
-    case validate_interactive_content_length(attrs) do
+    case contains_markdown_syntax?(attrs) do
       :ok ->
-        %InteractiveTemplate{}
-        |> InteractiveTemplate.changeset(attrs)
-        |> Repo.insert()
+        case validate_interactive_content_length(attrs) do
+          :ok ->
+            %InteractiveTemplate{}
+            |> InteractiveTemplate.changeset(attrs)
+            |> Repo.insert()
+
+          {:error, message} ->
+            {:error, message}
+        end
 
       {:error, message} ->
         {:error, message}
     end
   end
 
+  @spec contains_markdown_syntax?(map()) :: :ok | {:error, String.t()}
+  defp contains_markdown_syntax?(attrs) do
+    case check_interactive_content(attrs) do
+      :ok -> :ok
+      {:error, message} -> {:error, message}
```

Comment:
We can just return `check_interactive_content(attrs)`, no need of cases, if we are returning the same result as the function